dwave/plugins/sklearn/nearest_neighbors.pyx

```diff
@@ -0,0 +1,117 @@
+# distutils: language = c++
+import numpy as np
+cimport numpy as np
+from libcpp.vector cimport vector
+
+cdef extern from "math.h":
+    double log(double x) nogil
+
+def calculate_mi(np.ndarray[np.float_t,ndim = 2] X,
+                np.ndarray[np.float_t,ndim = 1] y,
+                unsigned long refinement_factor = 5): 
+        
+    cdef unsigned long n_obs = X.shape[0]
```

Comment:
In general, I have found it far less headache when going back and forth between NumPy and Cython to use fixed-width types everywhere. So
```suggestion
    cdef np.uint64_t n_obs = X.shape[0]
```

Cython, because everything gets moderated through an intermediate generated file, tends to get confused about typing, and keeping everything fixed tends to help.